### PR TITLE
Migrate from dotty to scala 3.3

### PR DESCRIPTION
**Pull request by rtar**
I removed some original Dotty-specific changes, so it might be worth to compare `src/main/scala-2` sources to `src/main/scala-3` sources for `Random.scala` instead of only looking at the changeset.

[I copy-pasted it to diffchecker for your convenience](https://www.diffchecker.com/kecWQGdp/)

### CODE REVIEW

File: src/main/scala-3/com/evolutiongaming/random/Random.scala [hunk 1]
```diff
@@ -17,19 +16,17 @@ object Random {
 
   type Seed = Long
 
-  // TODO: the pair of `RandomOps` + `given ops: RandomOps` could be replaced
-  // by simple `extenstion` block in latest Dotty versions, we only use this
-  // because we use Dotty 0.22 as it is the latest version that ScalaTest is
-  // available for, and we wanted to try out ScalaTest in Dotty
-  trait RandomOps {
-    def [F[_], G[_]](self: Random[F]) mapK (f: F ~> G): Random[G] = new Random[G] {
+  def apply[F[_]](implicit F: Random[F]): Random[F] = F
```
Comment (Z1kkurat) [inline src/main/scala-3/com/evolutiongaming/random/Random.scala:19]: Shouldn't this one be a `given`?

Comment (rtar) [reply src/main/scala-3/com/evolutiongaming/random/Random.scala]: I replaced `implicit` by `using`. I have no idea what is the idiomatic way to write this "summoning" pattern in Scala 3. May be there is an easier way?

**Review by Z1kkurat — APPROVED** [review]
LGTM, thanks!

File: src/main/scala-3/com/evolutiongaming/random/Random.scala (hunk 1 above)
Comment (Z1kkurat) [reply src/main/scala-3/com/evolutiongaming/random/Random.scala]: I'm not sure, let's leave it like this, maybe someone smarter than us will come later and fix it 😄

**Pull request merged**